File: Lab 3/Task2.c
#include <stdio.h>

int main (void)
{
  int a = 0, b = 0;

  puts ("Input 2 integers:");
  scanf ("%d %d", &a, &b);

  if (b == 0) {
    puts ("b can't be 0.");
    return 1;
  }

  printf (
    // FIXME: Overflow not handled
    "%d+%d=%d\n"
    "%d-%d=%d\n"
    "%d*%d=%d\n"
    "%d/%d=%d\n"
    "%d%%%d=%d\n",
    a, b, a + b,
    a, b, a - b,
    a, b, a * b,
    a, b, a / b,
    a, b, a % b
  );

  return 0;
}

File: Lab 3/Task2.lua
print("Input 2 integers:")
local buf_iter = string.gmatch(io.read(), "%d+")
local a, b = tonumber(buf_iter()), tonumber(buf_iter())

if b == 0 then
  print("b can't be 0.")
  return 1
end

io.write(string.format(
  "%d+%d=%d\n"..
  "%d-%d=%d\n"..
  "%d*%d=%d\n"..
  "%d/%d=%d\n"..
  "%d%%%d=%d\n",
  a, b, a + b,
  a, b, a - b,
  a, b, a * b,
  a, b, a // b, -- floor division
  a, b, a % b
))
